src/main/java/com/yelp/nrtsearch/server/luceneserver/search/collectors/DocCollector.java

```diff
@@ -186,8 +186,14 @@ <C extends Collector> CollectorManager<? extends Collector, SearcherResult> wrap
         request.getTerminateAfter() > 0
             ? request.getTerminateAfter()
             : indexState.getDefaultTerminateAfter();
+    int terminateAfterMaxRecallCount =
+        request.getTerminateAfterMaxRecallCount() > 0
+            ? request.getTerminateAfterMaxRecallCount()
+            : 0;
     if (terminateAfter > 0) {
-      wrapped = new TerminateAfterWrapper<>(wrapped, terminateAfter, () -> terminatedEarly = true);
+      wrapped =
+          new TerminateAfterWrapper<>(
+              wrapped, terminateAfter, terminateAfterMaxRecallCount, () -> terminatedEarly = true);
```

Comment:
When both terminate after and search timeout are used, the terminate after wraps the timeout. Do you want the timeout to always have priority? If so, this order may need to be reversed.